### PR TITLE
Remove MapboxNavigation dependency

### LINKED ISSUE #96: Remove dependency from `MapboxCoreNavigation`
## Description
`MapboxVisionAR` depends on `MapboxCoreNavigation` because there is an ability to set `NavigationService` to `VisionARViewController`. Downsides of this are: 
- need of synchronization with `MapboxCoreNavigation` releases
- consuming `NavigationService` which should belong to client application.

## Goal
Remove dependency from `MapboxCoreNavigation`

## Steps
- [ ] Review the API of `VisionARViewController` and remove all the dependencies on `MapboxCoreNavigation` framework.
- [ ] Move rerouting functionality to `demo` and internal applications

### LINKED ISSUE #116: Add Example of MapboxDirections.Route usage with vision ar api
After removing `MapboxNavigation` and `MApboxDirections` dependencies (issue #98) we wanna show users how to use these modules with our sdk. At least there should be added conversion from `MapboxDirections.Route` to `MapboxVisionARNative.Route`

### LINKED ISSUE #129: Make ios Examples buildable without private repositories
For now ios examples can't be built using carthage dependencies and its project settings without being a part of the entire workspace. So the `VisionExamples.xcodeproj`'s dependencies should be properly configured

### PR DESCRIPTION
**Pull request by IodaMikeMapbox**
The PR changes API that will break at least teaser app, that's why we should update teaser before merging

Checks:

* [x] Update changelog
* [x] Rebase to `dev` branch
* [x] Assign reviewers

Linked issues:
Resolves #96
Resolves #116 
Resolves #129

### CODE REVIEW

**Comment by dersim-davaod** [comment]
@IodaMikeMapbox please fix conflicts.

File: Cartfile.resolved [hunk 1]
```diff
@@ -1,10 +1,8 @@
 binary "https://www.mapbox.com/ios-sdk/Mapbox-iOS-SDK.json" "4.11.0"
```
Comment (chezzdev) [inline Cartfile.resolved]: Please remove all unnecessary dependencies brought by `MapboxNavigation`.

Comment (IodaMikeMapbox) [reply Cartfile.resolved]: ok

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,3 +3,6 @@
 ## 0.4.0 (In Development)
   - `VisionARManager` allows to change ARLane's length
   - `boundingBox` property on `MBVDetection` now stores normalized relative coordinates
+
+## 0.5.0 (Unreleased)
```
Comment (chezzdev) [inline CHANGELOG.md]: Please rebase. `## 0.5.0 - Unreleased` should be on the top.

File: MapboxVisionAR/Navigation/ARRoute+DirectionsRoute.swift [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#if canImport(MapboxDirections)
```
Comment (1ec5) [inline MapboxVisionAR/Navigation/ARRoute+DirectionsRoute.swift]: `canImport` doesn’t work with CocoaPods: https://github.com/mapbox/mapbox-navigation-ios/issues/1839#issuecomment-438789900.

File: Cartfile [hunk 4]
```diff
@@ -1,4 +1,3 @@
 binary "https://www.mapbox.com/ios-sdk/opencv.json" ~> 2.4.13
 github "mapbox/mapbox-events-ios" ~> 0.9.3
-github "mapbox/mapbox-navigation-ios" ~> 0.33.0
```
Comment (1ec5) [inline Cartfile]: By removing this explicit dependency, you also remove the ability to declare which versions the code is compatible with. canImport by itself doesn’t have a way to check for library versions. So the library is just as dependent on the navigation SDK as before, but without the safety that comes from the package manager checking for version compatibility.

File: MapboxVisionAR/Navigation/ARRoute+DirectionsRoute.swift (hunk 3 above)
Comment (IodaMikeMapbox) [reply MapboxVisionAR/Navigation/ARRoute+DirectionsRoute.swift]: yeah, we decided to remove it because of that

File: Cartfile (hunk 4 above)
Comment (IodaMikeMapbox) [reply Cartfile]: you right, it's a good reason to not use `canImport`

File: MapboxVisionAR/Render/VisionARViewController.swift [hunk 5]
```diff
@@ -29,18 +22,13 @@ public class VisionARViewController: UIViewController {
     }
 
     private var renderer: ARRenderer?
-    private var navigationManager: NavigationManager?
     
     /**
         Create an instance of VisionARNavigationController by specifying route controller from MapboxCoreNavigation framework.
```
Comment (dersim-davaod) [inline MapboxVisionAR/Render/VisionARViewController.swift]: Is doc comment correct?

Comment (chezzdev) [reply MapboxVisionAR/Render/VisionARViewController.swift]: Good catch.

Comment (IodaMikeMapbox) [reply MapboxVisionAR/Render/VisionARViewController.swift]: My bad, will fix

Comment (IodaMikeMapbox) [reply MapboxVisionAR/Render/VisionARViewController.swift]: done

**Comment by IodaMikeMapbox** [comment]
Teaser app updated, the pr won't break it

File: MapboxVision.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -207,6 +202,7 @@
 		2EDFFD72224BBD3E003BB718 /* ObservableVideoSource.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ObservableVideoSource.swift; sourceTree = "<group>"; };
 		2EF31C8F21904BB300752D23 /* RecordCoordinatorTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = RecordCoordinatorTests.swift; sourceTree = "<group>"; };
 		2EFF563E210B4A5B00B8D512 /* VideoTrimmer.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = VideoTrimmer.swift; sourceTree = "<group>"; };
+		5A780571229C3250005E6017 /* MapboxDirections.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = MapboxDirections.framework; path = Carthage/Build/iOS/MapboxDirections.framework; sourceTree = "<group>"; };
```
Comment (chezzdev) [inline MapboxVision.xcodeproj/project.pbxproj]: `MapboxDirections` and `MapboxCoreNavigation` are still under `MapboxVision/Frameworks` group.

Comment (IodaMikeMapbox) [reply MapboxVision.xcodeproj/project.pbxproj]: fixed

**Pull request merged**